src/ctl.c

```diff
@@ -1536,12 +1528,9 @@ thread_arena_ctl(tsd_t *tsd, const size_t *mib, size_t miblen, void *oldp,
 		}
 		/* Set new arena/tcache associations. */
 		arena_migrate(tsd, oldind, newind);
-		if (config_tcache) {
-			tcache_t *tcache = tsd_tcachep_get(tsd);
-			if (tcache != NULL) {
-				tcache_arena_reassociate(tsd_tsdn(tsd), tcache,
-				    newarena);
-			}
+		if (opt_tcache) {
```

Comment:
Yes that's right. tcache_available incorporates opt_tcache and should be the determining factor for auto tcache. Searched through the code a bit and the changes you mentioned sound complete to me.